release/pkg/assets/assets.go

```diff
@@ -0,0 +1,329 @@
+				}
+
+				if rc.DevRelease {
+					releaseName = fmt.Sprintf("%s-%s-eks-d-%s-%s-eks-a-%s-%s.%s",
+						archive.OSName,
+						kubeVersion,
+						eksDReleaseChannel,
+						eksDReleaseNumber,
+						rc.DevReleaseUriVersion,
+						arch,
+						imageExtensions[archive.Format],
+					)
+					releaseS3Path = fmt.Sprintf("artifacts/%s/eks-distro/%s/%s/%s-%s",
+						rc.DevReleaseUriVersion,
+						archive.Format,
+						eksDReleaseChannel,
+						eksDReleaseChannel,
+						eksDReleaseNumber,
+					)
+				} else {
+					releaseName = fmt.Sprintf("%s-%s-eks-d-%s-%s-eks-a-%d-%s.%s",
+						archive.OSName,
+						kubeVersion,
+						eksDReleaseChannel,
+						eksDReleaseNumber,
+						rc.BundleNumber,
+						arch,
+						imageExtensions[archive.Format],
+					)
+					releaseS3Path = fmt.Sprintf("releases/bundles/%d/artifacts/%s/%s", rc.BundleNumber, archive.Format, eksDReleaseChannel)
+				}
+			} else {
+				if archive.Format == "tarball" {
+					if rc.DevRelease || rc.ReleaseEnvironment == "development" {
+						sourceS3Key = fmt.Sprintf("%s-%s-%s-%s.tar.gz", archive.Name, os, arch, gitTag)
+						sourceS3Prefix = fmt.Sprintf("%s/%s", projectPath, latestPath)
+					} else {
+						sourceS3Key = fmt.Sprintf("%s-%s-%s.tar.gz", archive.Name, os, arch)
+						sourceS3Prefix = fmt.Sprintf("releases/bundles/%d/artifacts/%s/%s", rc.BundleNumber, archive.Name, gitTag)
+					}
+
+					if rc.DevRelease {
+						releaseName = fmt.Sprintf("%s-%s-%s-%s.tar.gz", archive.Name, rc.ReleaseVersion, os, arch)
+						releaseS3Path = fmt.Sprintf("artifacts/%s/%s/%s", rc.DevReleaseUriVersion, archive.Name, gitTag)
+					} else {
+						releaseName = fmt.Sprintf("%s-%s-%s.tar.gz", archive.Name, os, arch)
+						releaseS3Path = fmt.Sprintf("releases/bundles/%d/artifacts/%s/%s", rc.BundleNumber, archive.Name, gitTag)
+					}
+				}
+			}
+
+			cdnURI, err := utils.GetURI(rc.CDN, filepath.Join(releaseS3Path, releaseName))
+			if err != nil {
+				return nil, errors.Cause(err)
+			}
+
+			archiveArtifact := &releasetypes.ArchiveArtifact{
+				SourceS3Key:       sourceS3Key,
+				SourceS3Prefix:    sourceS3Prefix,
+				ArtifactPath:      filepath.Join(rc.ArtifactDir, fmt.Sprintf("%s-%s", archive.Name, archive.Format), eksDReleaseChannel, rc.BuildRepoHead),
+				ReleaseName:       releaseName,
+				ReleaseS3Path:     releaseS3Path,
+				ReleaseCdnURI:     cdnURI,
+				OS:                os,
+				OSName:            archive.OSName,
+				Arch:              []string{arch},
+				GitTag:            gitTag,
+				ProjectPath:       projectPath,
+				SourcedFromBranch: sourcedFromBranch,
+				ImageFormat:       archive.Format,
+			}
+
+			artifacts = append(artifacts, releasetypes.Artifact{Archive: archiveArtifact})
+		}
+	}
+
+	return artifacts, nil
+}
+
+func GetBundleReleaseAssets(supportedK8sVersions []string, eksDReleaseMap map[string]interface{}, rc *releasetypes.ReleaseConfig) (map[string][]releasetypes.Artifact, error) {
```

Comment:
The fact that we are using a map of empty interface here makes me think we might need to rethink this a bit

Reply:
agreed, we will be using a struct like
```go
type EksDLatestRelease struct {
    Dev         boolean `json:"dev"`
    Number      int     `json:"number`
    KubeVersion string  `json:"kubeVersion`
}
but the change to the releases file would need to happen first, as in the above comment.